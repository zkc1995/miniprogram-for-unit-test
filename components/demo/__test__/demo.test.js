const simulate = require('miniprogram-simulate');
const path = require('path')

test('i am demo test', () => {
  // 加载组件
  const id = simulate.load(path.join(__dirname, '../../demo/index'))
  // 创建组件实例
  const comp = simulate.render(id)
  // 创建一个父节点
  const parent = document.createElement('parent-wrapper')
  // 将组件依附在父节点上
  comp.attach(parent);
  // 获取子组件view
  const view = comp.querySelector('.index')
  // 预期渲染结果与结果是否一致
  // 情况1: 一致，则通过test、
  expect(view.dom.innerHTML).toMatchSnapshot()
  expect(view.dom.innerHTML).toBe('first-demo-for-unit-test')
  // 情况2: 不一致，则报错
  // expect(view.dom.innerHTML).toBe('first-demo-for')
  
  // 将组件从容器节点中移除
  comp.detach()
})