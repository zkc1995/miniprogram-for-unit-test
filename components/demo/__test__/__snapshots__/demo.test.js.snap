// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`i am demo test 1`] = `"first-demo-for-unit-test"`;
